### ISSUE TITLE
Crash when _dummy_ predicate is moved

### ISSUE THREAD

**Issue by peterstace** [dgraph-io/dgraph#1998]
```
2018/01/10 14:32:54 tablet.go:87: Going to move predicate _dummy_ from 1 to 2
2018/01/10 14:40:53 unexpected meta: 0
```
The "unexpected meta" error is fatal.

I added some debugging that showed that the relevant key is [0 0 7 95 100 117 109 109 121 95 0 0 0 0 0 0 0 0 0] (i.e. data key for `_dummy_`).

**Comment by peterstace** [comment]
Original approach to fixing this was to delete the `_dummy_` key rather than setting it, but this cause a badger crash (which should be investigated separately):

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc091f7]

goroutine 128666332 [running]:
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Item).Value(0x0, 0xc4200dc560, 0xc4200dc560, 0x1, 0x0, 0x19fb9a0)
	/home/petsta/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/iterator.go:80 +0x37
github.com/dgraph-io/dgraph/worker.movePredicateHelper(0x1a06020, 0xc4afdfb0e0, 0xc4c1789770, 0x7, 0x2, 0x0, 0x0)
	/home/petsta/go/src/github.com/dgraph-io/dgraph/worker/predicate_move.go:131 +0x69d
github.com/dgraph-io/dgraph/worker.(*grpcWorker).MovePredicate(0xc47e812020, 0x1a06020, 0xc4afdfb0e0, 0xc53888fe00, 0xc47e812020, 0x0, 0x0)
	/home/petsta/go/src/github.com/dgraph-io/dgraph/worker/predicate_move.go:277 +0x23b
github.com/dgraph-io/dgraph/protos/intern._Worker_MovePredicate_Handler(0x12a7b20, 0xc47e812020, 0x1a06020, 0xc4afdfb0e0, 0xc50ac92ff0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/petsta/go/src/github.com/dgraph-io/dgraph/protos/intern/internal.pb.go:2722 +0x276
google.golang.org/grpc.(*Server).processUnaryRPC(0xc443832300, 0x1a094c0, 0xc4be8da000, 0xc47e87d040, 0xc47e81a0c0, 0x19f0268, 0x0, 0x0, 0x0)
	/home/petsta/go/src/google.golang.org/grpc/server.go:877 +0xad6
google.golang.org/grpc.(*Server).handleStream(0xc443832300, 0x1a094c0, 0xc4be8da000, 0xc47e87d040, 0x0)
	/home/petsta/go/src/google.golang.org/grpc/server.go:1077 +0x1528
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4a3eceb70, 0xc443832300, 0x1a094c0, 0xc4be8da000, 0xc47e87d040)
	/home/petsta/go/src/google.golang.org/grpc/server.go:611 +0x9f
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/petsta/go/src/google.golang.org/grpc/server.go:609 +0xa1
```

**Comment by peterstace** [comment]
Fixed in master at bd3cba3
